README.ksonnet.md

```diff
@@ -0,0 +1,213 @@
+# Using Ksonnet 
+
+Kubeflow is currently evaluating different strategies for allowing deployments to be customized ([google/kubeflow/issues/23](https://github.com/google/kubeflow/issues/23)).
+
+This directory contains a prototype for using ksonnet to help us evaluate ksonnet.
+
+If you are unfamiliar with ksonnet you may want to start by reading the [tutorial](https://ksonnet.io/docs/tutorial)
+
+## Build ks
+
+You need a version of ksonnet newer than the [0.7.0 release](https://github.com/ksonnet/ksonnet/releases).
+
+As of this writing, Heptio hasn't released any newer prebuilt binaries so you will need to [build from source](https://ksonnet.io/docs/build-install).
+
+```
+# Clone the ksonnet repo into your GOPATH
+go get github.com/ksonnet/ksonnet
+cd ${GOPATH}/src/github.com/ksonnet/ksonnet
+make install
+```
+
+## Deploy Kubeflow
+
+Initialize a directory to contain your deployment
+```
+ks init my-kubeflow
+```
+
+Install the Kubeflow packages
+
+```
+cd my-kubeflow
+ks registry add kubeflow github.com/google/kubeflow/tree/master/kubeflow
+ks pkg install kubeflow/core
+ks pkg install kubeflow/tf-serving
+ks pkg install kubeflow/tf-job
+```
+
+
+Create the Kubeflow core component. The core component includes 
+  * JupyterHub
+  * TensorFlow job controller
+
+
+```
+ks generate core kubeflow-core --name=kubeflow-core --namespace=${NAMESPACE}
+```
+  * namespace is optional
+  * TODO(jlewi): There's an open github [issue](https://github.com/ksonnet/ksonnet/issues/222) to allow components to pull
+    the namespace from the environment namespace parameter.
+
+
+Define an environment that doesn't use any Cloud features
+  * This environment could be used for minikube or a full K8s cluster that doesn't depend on a cloud features.
+
+```
+ks env add nocloud
+```
+
+The default Kubeflow deployment will be suitable for this no cloud environment so you can just deploy the core components
+
+```
+ks apply nocloud -c kubeflow-core
+```
+
+If the user is running on a Cloud they could create an environment for this.
+
+```
+ks env add cloud
+ks param set --cloud=gke
+```
+   * The cloud parameter triggers a set of curated cloud configs.
```

Comment:
Describe what parameters this sets?

Reply:
Right now nothing, but it should control things like the CRD config for different clusters.